miniapps/multidomain/multidomain.cpp

```diff
@@ -0,0 +1,249 @@
+   cylinder_domain_attributes[0] = 1;
+
+   auto cylinder_submesh =
+      ParSubMesh::CreateFromDomain(mesh, cylinder_domain_attributes);
+
+   ParFiniteElementSpace h1_fes_cylinder(&cylinder_submesh, &h1_fec);
+
+   Array<int> inflow_attributes(cylinder_submesh.bdr_attributes.Max());
+   inflow_attributes = 0;
+   inflow_attributes[7] = 1;
+
+   Array<int> inner_cylinder_wall_attributes(
+      cylinder_submesh.bdr_attributes.Max());
+   inner_cylinder_wall_attributes = 0;
+   inner_cylinder_wall_attributes[8] = 1;
+
+   Array<int> inflow_tdofs, interface_tdofs, ess_tdofs;
+   h1_fes_cylinder.GetEssentialTrueDofs(inflow_attributes, inflow_tdofs);
+   h1_fes_cylinder.GetEssentialTrueDofs(inner_cylinder_wall_attributes,
+                                        interface_tdofs);
+   ess_tdofs.Append(inflow_tdofs);
+   ess_tdofs.Append(interface_tdofs);
+   ess_tdofs.Unique();
+   ConvectionDiffusionTDO cd_tdo(h1_fes_cylinder, ess_tdofs);
+
+   ParGridFunction temperature_cylinder_gf(&h1_fes_cylinder);
+   temperature_cylinder_gf = 0.0;
+   Vector temperature_cylinder;
+   temperature_cylinder_gf.GetTrueDofs(temperature_cylinder);
+
+   RK3SSPSolver cd_ode_solver;
+   cd_ode_solver.Init(cd_tdo);
+
+   Array<int> outer_domain_attributes(1);
+   outer_domain_attributes[0] = 2;
+
+   auto block_submesh = ParSubMesh::CreateFromDomain(mesh,
+                                                     outer_domain_attributes);
+
+   ParFiniteElementSpace h1_fes_block(&block_submesh, &h1_fec);
+
+   Array<int> block_wall_attributes(block_submesh.bdr_attributes.Max());
+   block_wall_attributes = 0;
+   block_wall_attributes[0] = 1;
+   block_wall_attributes[1] = 1;
+   block_wall_attributes[2] = 1;
+   block_wall_attributes[3] = 1;
+
+   Array<int> outer_cylinder_wall_attributes(
+      block_submesh.bdr_attributes.Max());
+   outer_cylinder_wall_attributes = 0;
+   outer_cylinder_wall_attributes[8] = 1;
+
+   h1_fes_block.GetEssentialTrueDofs(block_wall_attributes, ess_tdofs);
+
+   ConvectionDiffusionTDO d_tdo(h1_fes_block, ess_tdofs, 0.0, 1.0);
+
+   ParGridFunction temperature_block_gf(&h1_fes_block);
+   temperature_block_gf = 0.0;
+   ConstantCoefficient one(1.0);
+   temperature_block_gf.ProjectBdrCoefficient(one, block_wall_attributes);
+   Vector temperature_block;
+   temperature_block_gf.GetTrueDofs(temperature_block);
+
+   RK3SSPSolver d_ode_solver;
+   d_ode_solver.Init(d_tdo);
+
+   Array<int> cylinder_surface_attributes(1);
+   cylinder_surface_attributes[0] = 9;
+
+   auto cylinder_surface_submesh = ParSubMesh::CreateFromBoundary(mesh,
+                                                                  cylinder_surface_attributes);
+
+   ParFiniteElementSpace cylinder_surface_fes(&cylinder_surface_submesh, &h1_fec);
+   ParGridFunction cylinder_surface_gf(&cylinder_surface_fes);
+
+   double dt = 1.0e-5;
+   double t_final = 5.0;
+   double t = 0.0;
+   bool last_step = false;
+   int vis_steps = 10;
+
+   char vishost[] = "128.15.198.77";
```

Comment:
Maybe we should put 127.0.0.1 or localhost here.

Reply:
Yes, that's my local setup, sorry ;)